xtask/bench/src/lib.rs

```
@@ -1,90 +1,76 @@
-use ansi_rgb::{red, Foreground};
-use itertools::Itertools;
-use rslint_errors::Diagnostic;
+mod features;
+mod utils;
+
 use std::fmt::{Display, Formatter};
-use std::ops::Add;
+use std::str::FromStr;
 use std::time::Duration;
-use std::{path::PathBuf, str::FromStr};
 
-fn err_to_string<E: std::fmt::Debug>(e: E) -> String {
-    format!("{:?}", e)
+pub use crate::features::formatter::benchmark_format_lib;
+use crate::features::formatter::BenchmarkFormatterResult;
+pub use crate::features::parser::benchmark_parse_lib;
+use crate::features::parser::BenchmarkParseResult;
+pub use utils::get_code;
+
+/// What feature to benchmark
+pub enum FeatureToBenchmark {
+    /// benchmark of the parser
+    Parser,
+    /// benchmark of the formatter
+    Formatter,
+}
+
+impl FromStr for FeatureToBenchmark {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "parser" => Ok(Self::Parser),
+            "formatter" => Ok(Self::Formatter),
+            _ => Ok(Self::Parser),
+        }
+    }
 }
 
-#[cfg(feature = "dhat-on")]
-fn print_diff(before: dhat::Stats, current: dhat::Stats) -> dhat::Stats {
-    use humansize::{file_size_opts as options, FileSize};
-
-    println!("\tMemory");
-    if let Some(heap) = &current.heap {
-        println!("\t\tCurrent Blocks: {}", heap.curr_blocks);
-        println!(
-            "\t\tCurrent Bytes: {}",
-            heap.curr_bytes.file_size(options::CONVENTIONAL).unwrap()
-        );
-        println!("\t\tMax Blocks: {}", heap.max_blocks);
-        println!(
-            "\t\tMax Bytes: {}",
-            heap.max_bytes.file_size(options::CONVENTIONAL).unwrap()
-        );
+impl Display for FeatureToBenchmark {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let _ = match self {
```

Why not return the error instead of ignoring it ?